shared/src/sources/balancer/info_fetching.rs

```diff
@@ -82,11 +86,55 @@ impl PoolInfoFetching for PoolInfoFetcher {
             .into_iter()
             .map(Bfp::from_wei)
             .collect();
-        Ok(WeightedPoolInfo {
+        Ok(PoolInfo {
             pool_id,
             tokens,
             weights,
             scaling_exponents,
+            amplification_parameter: Default::default(),
+        })
+    }
+
+    async fn get_stable_pool_data(&self, pool_address: H160) -> Result<PoolInfo> {
```

Comment:
Can we write a test for this using the new contract mocking feature introduced by @taminomara ?

Reply:
Sure thing. it will take me a bit to get used to it.

Reply:
I think that we are still going to run into the issue of having a mockable `CallBatch` and `Web3`for these methods... Is this something you were aware of when making this suggestion?

Reply:
Yes, with the latest `ethcontract` release, we should have mockable contracts and mock Web3 that should allow testing. Maybe @taminomara might be able to give you some pointers when you're back from vacation.

Reply:
All tested and ready to rock!